File: Rectangle_Rotation.java
class Solution {

    static int rectangleRotation(final int a, final int b) {
    
        int sqrtA = (int) Math.floor(a / Math.sqrt(2));
        int sqrtB = (int) Math.floor(b / Math.sqrt(2));
        
        if (sqrtA % 2 != 0 ^ sqrtB % 2 != 0) {
            return (sqrtA + 1) * (sqrtB + 1) + sqrtA * sqrtB - 1;    
        }
        
        return (sqrtA+1) * (sqrtB+1) + sqrtA * sqrtB;
    }
}

File: Rectangle_Rotation.py
import math

def rectangle_rotation(a, b):
  sqrt_a = int(a / math.sqrt(2))
  sqrt_b = int(b / math.sqrt(2))
      
  return (sqrt_a + 1) * (sqrt_b + 1) + sqrt_a * sqrt_b - (1 if (sqrt_a % 2 != 0) ^ (sqrt_b % 2 != 0) else 0)
